Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp

```diff
@@ -117,16 +134,31 @@ static unsigned computeStructTypeHash(size_t fieldCount, const StructField* fiel
     return accumulator;
 }
 
+static unsigned computeArrayTypeHash(FieldType elementType)
+{
+    unsigned accumulator = 0x7835ab;
+    accumulator = WTF::pairIntHash(accumulator, WTF::IntHash<uint8_t>::hash(static_cast<uint8_t>(elementType.type.kind)));
+    accumulator = WTF::pairIntHash(accumulator, WTF::IntHash<uint8_t>::hash(static_cast<uint8_t>(elementType.type.nullable)));
+    accumulator = WTF::pairIntHash(accumulator, WTF::IntHash<TypeIndex>::hash(elementType.type.index));
```

Comment:
Are type indexes unique? As in, the type index is correlated with kind and mutability? Why not just hash type index + module?

Reply:
In this case no, it's not unique. In particular, the null type index is used for a lot of types. i.e., all non-reftypes like `I32`. Even if the type index is non-null, the mutability can vary and the kind can be either `Ref` or `RefNull`.